tests/testthat/test_plots.R

```diff
@@ -3,95 +3,173 @@ context("Basic time series graphs")
+    expect_warning(plot(t, xlim = lubridate::ymd(c(20050101, 20101231))))
+    x <- inzightts(visitorsQ, var = 2:5)
+    expect_error(plot(x, c("Australia", "Japan"), ylab = "AUS"))
+    expect_warning(plot(x, names(x)[-1], aspect = 4))
+    expect_is(plot(t, aspect = 1), "ggplot")
+    expect_equal(guess_plot_var(t, "test"), c("", "test"))
+    expect_is(guess_plot_var(t, NULL), "name")
+    expect_is(plot(y), "ggplot")
+    expect_is(plot(y, compare = FALSE), "ggplot")
+    expect_is(plot(y, emphasise = 2), "ggplot")
+    expect_is(plot(y, emphasise = 2, non_emph_opacity = 1e-6), "ggplot")
+    expect_is(plot(y, emphasise = 2, non_emph_opacity = 0), "ggplot")
+    x <- visitorsM2 %>%
+        dplyr::mutate(y = ggplot2::cut_number(Japan, 4)) %>%
+        inzightts()
+    y <- visitorsA2 %>%
+        tidyr::pivot_longer(!Time, names_to = "Country", values_to = "Visitors") %>%
+        dplyr::mutate(y = ggplot2::cut_number(Visitors, 4)) %>%
+        inzightts(key = "Country")
+    expect_is(plot(x, "y"), "ggplot")
+    # TODO: spitting out a warning
+    #   Key detected when plotting category, setting `emphasise = 1L`
+    expect_is(plot(y, "y"), "ggplot")
+    expect_is(plot(y, "y"), "ggplot")
+    expect_is(plot(x, "y", pal = 1:4), "ggplot")
+    expect_is(plot(x, "y", pal = 1:5), "ggplot")
+    # TODO: spitting out a warning (same as above)
+    expect_is(plot(y, "y", pal = 1:5), "ggplot")
+    expect_error(plot(y, c("y", "Visitors")))
+    expect_warning(plot(y, c("y", "y")))
+    expect_warning(plot(x, "y", aspect = 1))
+    x$Australia[c(23, 25)] <- NA
+    # TODO: spitting out warnings
+    #   1: Time gaps detected, STL returning NULL model.
+    #   2: Time gaps in all (key) levels, turning off smoothers.
+    expect_is(plot(x), "ggplot")
+})
 
+test_that("Check decomposition plot configuration", {
+    expect_error(decomp(t, sm_model = "null_model"))
+    expect_is(iNZightTS2:::use_decomp_method("stl"), "use_stl")
+    expect_is(iNZightTS2:::use_decomp_method("test"), "use_test")
+    expect_equal(iNZightTS2:::as_year.character("2010"), 2010)
+    expect_equal(iNZightTS2:::as_year.vctrs_vctr(tsibble::yearquarter("2010Q1")), 2010)
+    expect_equal(iNZightTS2:::as_year.numeric(2010), 2010)
+    expect_true(fabletools::is_dable(
+        .decomp(
+            iNZightTS2:::use_decomp_method("stl"),
+            t, "Australia",
+            mult_fit = TRUE
+        )
+    ))
+    t_gap <- t
+    t_gap$Australia[3] <- NA
+    expect_warning(decomp(t_gap, "Australia"))
+})
 
+test_that("Check season plot configuration", {
+    expect_is(seasonplot(t, labels = "none"), "patchwork")
+    expect_error(seasonplot(t, model_range = 2001))
+    # TODO: does not produce a warning
+    expect_warning(seasonplot(t, model_range = lubridate::ymd(c(20050101, 20101231))))
+    expect_is(seasonplot(tm, names(tm)[-1]), "patchwork")
+    t_key_gap <- y
+    t_key_gap$Australia[3] <- NA
+    # TODO: gives ERROR instead of WARNING
```

Comment:
Not sure why test these tests were successful for me without returning an error. To be discussed.